org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/CombinedBinaryExpression.java

```diff
@@ -236,6 +238,50 @@ public void generateOptimizedStringConcatenation(BlockScope blockScope,
 	}
 }
 
+@Override
+public void buildStringForConcatation(BlockScope blockScope, CodeStream codeStream, int typeID, StringBuilder recipe, List<TypeBinding> argTypes) {
+	if (this.referencesTable == null) {
+		super.buildStringForConcatation(blockScope, codeStream,	typeID, recipe, argTypes);
```

Comment:
Can you please check to make sure that we have 100% branch coverage for the new code in Expression, BinaryExpression and CombinedBinaryExpression

Reply:
> Can you please check to make sure that we have 100% branch coverage for the new code in Expression, BinaryExpression and CombinedBinaryExpression

@jarthana : this review comment perhaps fell through the cracks. Without the new test added by @iloveeclipse viz `org.eclipse.jdt.core.tests.compiler.regression.StringConcatTest.test005()` in the context of https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1202 there is no test in the full run of `RunJDTCoreTests` that exercises that branch :-(

Here is what I suggest/would do: In `Expression`, `BinaryExpression` and `CombinedBinaryExpression`, in all the new code, I would insert a `throw new UnsupportedOperationException();` and comment out the code there. Now `RunJDTCoreTests` to make sure that every one of those exception actually triggers.

Reply:
I'm running the suite in HEAD/master with coverage / EclEmma enabled. Will report the results here.

Reply:
results:
* Expression.buildStringForConcatation() not covered: lines 965-7 (block starting codeStream.ldc)
* BinaryExpression.buildStringForConcatation() not covered: line 1572 (super call)
   * first compound if condition didn't see all combinations
* CombinedBinaryExpression.buildStringForConcatation() not covered:
  * lines 261/2, this was [#1202](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1202) (tested before that one got merged)
  * lines 265-7  (inner else block)
  * line 281 (final super call)

Reply:
BTW, do you care to fix the spelling of concatenation in the method name?